src/plugins/mf_priority.cpp

```diff
@@ -307,6 +307,128 @@ static int validate_cb (flux_plugin_t *p,
+    int max_run_jobs, cur_active_jobs, max_active_jobs = 0;
+    double fairshare = 0.0;
+    struct bank_info *b;
+
+    std::map<int, std::map<std::string, struct bank_info>>::iterator it;
+    std::map<std::string, struct bank_info>::iterator bank_it;
+
+    flux_t *h = flux_jobtap_get_flux (p);
+    if (flux_plugin_arg_unpack (args,
+                                FLUX_PLUGIN_ARG_IN,
+                                "{s:i, s{s{s{s?s}}}}",
+                                "userid", &userid,
+                                "jobspec", "attributes", "system",
+                                "bank", &bank) < 0) {
+        return flux_jobtap_reject_job (p, args, "unable to unpack bank arg");
+    }
+
+    b = static_cast<bank_info *> (flux_jobtap_job_aux_get (
+                                                    p,
+                                                    FLUX_JOBTAP_CURRENT_JOB,
+                                                    "mf_priority:bank_info"));
+
+    if (b != NULL) {
+        max_run_jobs = b->max_run_jobs;
+        fairshare = b->fairshare;
+        cur_active_jobs = b->cur_active_jobs;
+        max_active_jobs = b->max_active_jobs;
+    } else {
+        // make sure user belongs to flux-accounting DB
+        it = users.find (userid);
+        if (it == users.end ())
+            // user does not exist in internal map yet, so create a bank_info
+            // struct that signifies it's going to be held in PRIORITY
+            goto no_user_info;
+
+        // make sure user belongs to bank they specified; if no bank was passed
+        // in, look up their default bank
+        if (bank != NULL) {
+            bank_it = it->second.find (std::string (bank));
+            if (bank_it == it->second.end ())
+                return flux_jobtap_reject_job (p, args,
+                                      "user does not belong to specified bank");
+        } else {
+            bank = const_cast<char*> (users_def_bank[userid].c_str ());
+            bank_it = it->second.find (std::string (bank));
+            if (bank_it == it->second.end ())
+                return flux_jobtap_reject_job (p, args,
+                                      "user/default bank entry does not exist");
+        }
+
+        max_run_jobs = bank_it->second.max_run_jobs;
+        fairshare = bank_it->second.fairshare;
+        cur_active_jobs = bank_it->second.cur_active_jobs;
+        max_active_jobs = bank_it->second.max_active_jobs;
+
+        b = &bank_it->second;
+    }
+
+    // if a user's fairshare value is 0, that means they shouldn't be able
+    // to run jobs on a system
+    if (fairshare == 0)
+        return flux_jobtap_reject_job (p, args, "user fairshare value is 0");
+
+    // if a user/bank has reached their max_active_jobs limit, subsequently
+    // submitted jobs will be rejected
+    if (max_active_jobs > 0 && cur_active_jobs >= max_active_jobs)
+        return flux_jobtap_reject_job (p, args, "user has max active jobs");
```

Comment:
Ah, I see. Yeah, I think that, for the most part, I just duplicated what I had in `job.validate`, which is why the `flux_jobtap_reject_job ()` calls are in here, so I don't think these lines are covered by tests so far, sorry about that. 🤦‍♂️

> Maybe we should discuss if we should address that issue in flux-core before merging this change, if it makes your work here more straightforward?

I think that sounds like a good plan. I wonder if it would be okay to instead raise a job exception here in `job.new` instead of rejecting the job?

Reply:
Yeah, that might be reasonable for now.